src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -233,19 +236,23 @@ public void doIteration() {
             } catch (IOException e) {
                 String warning = "Cannot connect to instance " + instance + ". Is a JMX Server running at this address?";
                 LOGGER.warn(warning);
-                reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
+                this.reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
+                this.sendServiceCheck(reporter, instance, warning, Status.STATUS_ERROR);
             } catch (SecurityException e) {
                 String warning = "Cannot connect to instance " + instance + " because of bad credentials. Please check your credentials";
                 LOGGER.warn(warning);
-                reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
+                this.reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
+                this.sendServiceCheck(reporter, instance, warning, Status.STATUS_ERROR);
             } catch (FailedLoginException e) {
                 String warning = "Cannot connect to instance " + instance + " because of bad credentials. Please check your credentials";
                 LOGGER.warn(warning);
-                reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
+                this.reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
+                this.sendServiceCheck(reporter, instance, warning, Status.STATUS_ERROR);
             } catch (Exception e) {
                 String warning = "Cannot connect to instance " + instance + " for an unknown reason." + e.getMessage();
                 LOGGER.fatal(warning, e);
-                reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
+                this.reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
+                this.sendServiceCheck(reporter, instance, warning, Status.STATUS_ERROR);
             }
         }
```

Comment:
All of these exceptions are pretty much handled the same way, only the `warning` message differs. I am wondering if we can factorize here:
``` java

try {
	doSomething();
} catch (Exception e) {
	String warning = null;

	if (e instanceof SomeException) {
		warning = "a scary warning"

	} else if( e instanceof AnOtherException){
		warning = "some warning"

	} else {
		// You got the idea
		(...)
	}


	this.reportStatus(appConfig, reporter, instance, 0, warning, Status.STATUS_ERROR);
	this.sendServiceCheck(reporter, instance, warning, Status.STATUS_ERROR);
}
```